components/legacy_team_sidebar/components/team_button.tsx

```diff
@@ -1,77 +1,65 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 import {Tooltip} from 'react-bootstrap';
-import {injectIntl} from 'react-intl';
+import {injectIntl, IntlShape} from 'react-intl';
 import {Link} from 'react-router-dom';
 import {Draggable} from 'react-beautiful-dnd';
 import classNames from 'classnames';
 
 import {mark, trackEvent} from 'actions/diagnostics_actions.jsx';
 import Constants from 'utils/constants';
-import {intlShape} from 'utils/react_intl';
 import {isDesktopApp} from 'utils/user_agent';
-import {localizeMessage, isMac} from 'utils/utils.jsx';
+import {isMac, localizeMessage} from 'utils/utils.jsx';
 import CopyUrlContextMenu from 'components/copy_url_context_menu';
 import OverlayTrigger from 'components/overlay_trigger';
 import TeamIcon from '../../widgets/team_icon/team_icon';
 
+interface Props {
+    btnClass?: string;
+    url: string;
+    displayName?: string;
+    content?: string;
+    tip: string | JSX.Element;
+    order?: number;
+    showOrder?: boolean;
```

Comment:
This prop should be required.

Reply:
Doing this throws the error I have mentioned above also mentioning again
```
error TS2740: Type '{ btnClass: string; key: string; url: string; tip: Element; content: string; switchTeam: Function; }' is missing the following properties from type 'Pick<Props, "content" | "disabled" | "displayName" | "url" | "placement" | "btnClass" | "active" | "teamId" | "tip" | "order" | "showOrder" | "unread" | "mentions" | "teamIconUrl" | "switchTeam" | "isDraggable" | "teamIndex">': disabled, displayName, placement, active, and 8 more.

```

Reply:
Which change? And where does this error pop up?

Reply:
The change to add optional in every props. It is throwing error in the following lines
line [219](https://github.com/Ayanrocks/mattermost-webapp/blob/legacy_team_sidebar-ts/components/legacy_team_sidebar/legacy_team_sidebar_controller.tsx#L219-L234)
line [242](https://github.com/Ayanrocks/mattermost-webapp/blob/legacy_team_sidebar-ts/components/legacy_team_sidebar/legacy_team_sidebar_controller.tsx#L242-L254)
line [262](https://github.com/Ayanrocks/mattermost-webapp/blob/legacy_team_sidebar-ts/components/legacy_team_sidebar/legacy_team_sidebar_controller.tsx#L262-L273)
 of the legacy_team_sidebar_controller

here is the full error for better understanding

```shell
> mattermost-webapp@0.0.1 check-types /home/ayanrocks/code/mattermost-webapp
> tsc

components/legacy_team_sidebar/legacy_team_sidebar_controller.tsx:219:18 - error TS2741: Property 'content' is missing in type '{ key: string; url: string; tip: string; active: boolean; displayName: string; unread: boolean; order: number; showOrder: boolean; mentions: number; teamIconUrl: string | null; switchTeam: (url: string) => (dispatch: Dispatch<...>, getState: GetStateFunc) => void; isDraggable: true; teamId: string; teamIndex: number...' but required in type 'Pick<Props, "content" | "disabled" | "displayName" | "url" | "placement" | "btnClass" | "active" | "teamId" | "tip" | "order" | "showOrder" | "unread" | "mentions" | "teamIconUrl" | "switchTeam" | "isDraggable" | "teamIndex">'.

219                 <TeamButton
                     ~~~~~~~~~~

  components/legacy_team_sidebar/components/team_button.tsx:23:5
    23     content: string;
           ~~~~~~~
    'content' is declared here.

components/legacy_team_sidebar/legacy_team_sidebar_controller.tsx:242:18 - error TS2739: Type '{ btnClass: string; key: string; url: string; tip: Element; content: string; switchTeam: (url: string) => (dispatch: Dispatch<GenericAction>, getState: GetStateFunc) => void; }' is missing the following properties from type 'Pick<Props, "content" | "disabled" | "displayName" | "url" | "placement" | "btnClass" | "active" | "teamId" | "tip" | "order" | "showOrder" | "unread" | "mentions" | "teamIconUrl" | "switchTeam" | "isDraggable" | "teamIndex">': order, showOrder

242                 <TeamButton
                     ~~~~~~~~~~

components/legacy_team_sidebar/legacy_team_sidebar_controller.tsx:262:22 - error TS2739: Type '{ btnClass: string; url: string; tip: Element; content: string; switchTeam: (url: string) => (dispatch: Dispatch<GenericAction>, getState: GetStateFunc) => void; }' is missing the following properties from type 'Pick<Props, "content" | "disabled" | "displayName" | "url" | "placement" | "btnClass" | "active" | "teamId" | "tip" | "order" | "showOrder" | "unread" | "mentions" | "teamIconUrl" | "switchTeam" | "isDraggable" | "teamIndex">': order, showOrder

262                     <TeamButton
                         ~~~~~~~~~~



```

Reply:
Ahh okay, looks like it's not really required but marked as such. This is one of those weird ones where it probably should be required, but there's no real impact on the program flow if it's not provided. Let's keep it optional then. Sorry about the holdup.